orchestrator/services/watch_service.py

```
@@ -0,0 +1,601 @@
+
+        watch.status = new_status.value
+        if new_status in TERMINAL_WATCH_STATUSES:
+            watch.closed_at = _utcnow()
+        elif current in TERMINAL_WATCH_STATUSES:
+            # escalated -> watching renewal re-opens the watch
+            watch.closed_at = None
+
+        try:
+            db.flush()
+        except StaleDataError:
+            raise ConflictError(entity_type="watch", entity_id=watch.id)
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)